commons-jcs-core/src/main/java/org/apache/commons/jcs3/log/LogManager.java

```diff
@@ -57,15 +61,34 @@ private static LogFactory createLogFactory()
                 LogManager.logSystem = System.getProperty("jcs.logSystem",
                         LOGSYSTEM_JAVA_UTIL_LOGGING);
             }
-
-            for (final LogFactory factory : factories)
-            {
-                if (logSystem.equalsIgnoreCase(factory.getName()))
-                {
-                    return factory;
+            List<ServiceConfigurationError> errors = new ArrayList<>();
+            Iterator<LogFactory> itr = factories.iterator();
+            LogFactory factory = null;
+            while (itr.hasNext()) {
+                try {
+                    LogFactory instance = itr.next();
+                    if (logSystem.equalsIgnoreCase(instance.getName())) {
+                        factory = instance;
+                        break;
+                    }
+                } catch (ServiceConfigurationError e) {
+                    errors.add(e);
                 }
             }
-
+            if (factory != null) {
+                if (!errors.isEmpty()) {
+                    Log log = factory.getLog(LogFactoryHolder.class);
```

Comment:
What is LogFactoryHolder? What kind of log system do you use here? I think I understand your intention but we have like a chicken-and-egg problem here. I'd rather not instantiate a logger while trying to select a log system. WDYT?

Reply:
At this point - we have already selected the logging system (i.e. factory is not null). I just added logging showing the issues with loading other loggers in debug.

Reply:
`LogFactoryHolder` is an inner class within `LogManager`. We could use `LogManager` instead of `LogFactoryHolder` - I just defaulted to the leaf.

Reply:
Yeah, you are right. My bad.